CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+# Contributuing
+
+Contributions to this project are welcome, whether minor updates to documentation or entirely new features.
+This document outlines the technical and procedural requirements for your work to be included.
+All contributions, in any form, must also confirm with the [code of conduct](CODE_OF_CONDUCT.md).
+
+> :warning: This document is a living document, and will adapt based on the project's needs.
+> If some part of this document creates a hindrance, raise an issue explaining the problem.
+
+## Scope: What Lives in this Repository?
+
+MewBot's core design includes the tenant of modularity.
+The core repo will only accept features and components that do not add additional dependencies.
```

Comment:
I mean, once we stabilise what the requirements actually are ...

Definitely aiohttp. Probably some kind of file handling library. Perhaps a decent async database framework for the object stores?

Reply:
I have reworded this section to be positively worded -- it now explains what is part of the core functionality, and that adding third party dependencies requires extra consideration.

Reply:
I agree with the rewording - both the sentiment and the specific language used.

In practise, I think we're going to struggle with this - especially as we add more io options. If we can keep it down to less than 1 per io type, I'll be personally happy. But there's every reason to have this language in the guide as a target.

This is partially resolvable by a more sophisticated build/install system.